lib/fluent/plugin/in_tail.rb

```diff
@@ -244,8 +247,9 @@ def close_watcher(tw, close_io = true)
     end
 
     def close_watcher_after_rotate_wait(tw)
-      closer = TailWatcher::Closer.new(@rotate_wait, tw, log, &method(:close_watcher))
-      closer.attach(@loop)
+      timer_execute(:in_tail_close_watcher, @rotate_wait, repeat: false) do
```

Comment:
Yes, this code calls `close_watcher(tw)` at once.
Because `repeat: false` is specified, see also ev(3).